lib/timeline.js

```diff
@@ -0,0 +1,69 @@
+const guid = require('./utils/guid')
+
+class Timeline {
+  constructor(client) {
+    this.client = client
+  }
+
+  createEventType(applicationId, userId, data, cb) {
+    data['applicationId'] = data['applicationId'] || applicationId
+    return this.client._request(
+      {
+        method: 'POST',
+        path: `/integrations/v1/${applicationId}/timeline/event-types?userId=${userId}`,
+        body: data,
+      },
+      cb,
+    )
+  }
+
+  updateEventType(eventTypeId, applicationId, data, cb) {
+    data['applicationId'] = data['applicationId'] || applicationId
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventTypeId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createEventTypeProperty(eventTypeId, applicationId, userId, data, cb) {
+    const parameters = {
+      method: 'POST',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventTypeId}/properties?userId=${userId}`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  updateEventTypeProperty(eventTypeId, applicationId, propertyId, data, cb) {
+    data['id'] = data['id'] || propertyId
+    const parameters = {
+      method: 'PUT',
+      path: `/integrations/v1/${applicationId}/timeline/event-types/${eventTypeId}/properties`,
+      body: data,
+    }
+
+    return this.client._request(parameters, cb)
+  }
+
+  createTimelineEvent(applicationId, eventTypeId, data, cb) {
+    if (!data.id) {
```

Comment:
I'm not certain which `id` we're looking for in this case and I'm curious about setting the `id` value to `guid()` since this is a create action. Is the `id` relevant to the Timeline?

Reply:
The id is required by HubSpot... I think we're using a guid here to ensure it produces something unique.

Reply:
Leaving this here for others (pulled from the [HubSpot API docs](https://developers.hubspot.com/docs/methods/timeline/create-or-update-event)):

> The ID of the event you're creating.  You will need to specify your own event ID when creating an ID, it will not be assigned automatically.